duckduckgo/Android
# Feature/consolidated landing page

Task/Issue URL: https://app.asana.com/0/471012189846020/487243693211914
Tech Design URL:
CC:

**Description**:
Users provided feedback that they didn't like the two-step UI when performing a search. This task removes that interim view which showed the DDG logo, and instead consolidates it with the existing browser activity.

Note, there is an outstanding question around whether to show the 🔥button while the user is focussed on the omnibar, or when typing, or just always. This implements the "show always" approach for now until clarification is offered on this.

**Steps to test this PR**:
1. Launch the app; verify the UI looks good and responds well to the keyboard being shown/hidden. The DDG logo should remain nicely visible when there is nothing loaded in the `WebView`.
1. Test the 🔥 button from empty state. Test 🔥button from state when `WebView` is populated.
1. Test the app's internal navigation links, including bookmarks, TOSDR, About DDG etc...
1. Test the widget; this should always launch a "fresh" search view - DDG logo shown with the user ready to start typing
1. Test setting the app as the default Assist app (see #143); this should also launch the app with a "fresh" search view

⚠️ Requires changes in quite a few areas since we have a variety of ways of launching pages / the app (including internally from pages like the About DuckDuckGo activity, as well as externally using the Assist intent, the widget etc...)

As such, a **thorough smoke test is requested** as part of testing this PR since it really touches all areas.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
